src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -237,6 +262,117 @@ QString MerHardwareDeviceWizardSelectionPage::detectDeviceName(
     return deviceName;
 }
 
+bool MerHardwareDeviceWizardSelectionPage::detectSdkClientTools(
+        const SshConnectionParameters &sshParams, bool *ok)
+{
+    if (!*ok) {
+        return false;
+    }
+
+    SshRemoteProcessRunner runner;
+    QEventLoop loop;
+    connect(&runner, &SshRemoteProcessRunner::connectionError,
+            &loop, &QEventLoop::quit);
+    connect(&runner, &SshRemoteProcessRunner::processClosed,
+            &loop, &QEventLoop::quit);
+    runner.run("sed -n '/^VERSION_ID=/s/^VERSION_ID=\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*/\\1/p'"
+               " /etc/os-release", sshParams);
+    loop.exec();
+
+    if (!runner.lastConnectionErrorString().isEmpty()
+            || runner.processExitStatus() != SshRemoteProcess::NormalExit
+            || runner.processExitCode() != 0) {
+        *ok = false;
+        qWarning() << "Failed to query os version name on target";
+        return false;
+    }
+
+    const QString osVersion = QString::fromLatin1(runner.readAllStandardOutput()).trimmed();
+    if (osVersion.isEmpty()) {
+        *ok = false;
+        qWarning() << "Empty output from querying os version name on target";
```

Comment:
qCWarning

Reply:
Changed all of the qWarnings in that file to use it, I hope that's fine.

Reply:
Could you please change it in all files under src/plugins/mer then? There is just few occurrences.

```
merconnectionmanager.cpp:        qWarning() << "VirtualMachine::update() - unknown state";
mersdkmanager.cpp:            qWarning() << "MerToolChain without target found. Removing toolchain.";
mersdkmanager.cpp:            qWarning() << "MerQtVersion without target found. Removing qtversion.";
mersdkmanager.cpp:                qWarning() << "Broken Mer kit found! Removing kit.";
mersdkmanager.cpp:        qWarning() << "Sysroot does not exist" << buildTarget.sysRoot.toString();
mersettings.cpp:        qWarning() << "The environment variable"

```

In `merconnectionmanager.cpp` you may rather remove the `default:` case with the warning. This is checked by the compiler nowadays.

Reply:
Sure.